clustering.Rmd

```diff
@@ -541,57 +662,65 @@ penguin_clust_ks <- tibble(k = 1:9)
 penguin_clust_ks
 ```
 
-Then we use `map` \index{map} to apply the `kmeans` function to each
-K. However, we need to use `map` a little bit differently than we have
-before. This is because we need to iterate over `k`, which is the *second argument* 
-to the `kmeans` function. In the past, we have used `map` only to iterate over values of the
-*first argument* of a function. Since that is the default, we could simply 
-write `map(data_frame, function_name)`. This won't work here; we need to 
-provide our data frame as the first argument to the `kmeans` function.
-
-The solution is to create something called an *anonymous function*. \index{function!anonymous}
-An anonymous function is a function that has no name,
-unlike other functions you've seen so far (`kmeans`, `select`, etc). 
-To do this we will write our map statement like this:
-
-```{r, eval = FALSE}
-map(penguin_clust_ks, function(k) kmeans(scaled_data, k))
+Then we use `rowwise` \index{rowwise} + `mutate` to apply the `kmeans` function 
+within each row, to each K. 
+However, given that the `kmeans` function 
+returns a model object to us, (not a vector),
+we will need to store the results as a list columm.
+This can work because lists are super vectors, and so they are legitimate 
```

Comment:
"super vectors" -- I know you did some work on wrangling. Do we use the term "super vector" there? I've never heard this term before...

Reply:
fixed.